cmd/endpoint.go

```diff
@@ -65,18 +65,20 @@ func (endpoint Endpoint) Type() EndpointType {
 	return URLEndpointType
 }
 
-// SetHTTPS - sets secure http for URLEndpointType.
-func (endpoint Endpoint) SetHTTPS() {
+// IsHTTPS - returns true if secure for URLEndpointType.
+func (endpoint Endpoint) IsHTTPS() bool {
 	if endpoint.Host != "" {
-		endpoint.Scheme = "https"
+		return endpoint.Scheme == "https"
 	}
+	return false
 }
 
-// SetHTTP - sets insecure http for URLEndpointType.
-func (endpoint Endpoint) SetHTTP() {
+// IsHTTP - returns true if insecure for URLEndpointType.
+func (endpoint Endpoint) IsHTTP() bool {
 	if endpoint.Host != "" {
-		endpoint.Scheme = "http"
+		return endpoint.Scheme == "http"
 	}
+	return false
```

Comment:
I see you are moving `SetHTTP()` and `SetHTTPS()` methods.  You can remove them completely and add only one method `IsHTTPS()` i.e. `IsHTTP()` is negative duplicate.